### PR TITLE
Expanding hints scope

### PR DESCRIPTION
**Pull request by fileoy**
Reference to issue #1148 :

- Now it accepts all hint types from PubTator rather than just organisms. e.g. genes, cell lines, diseases, chemicals
- Added tests for the new hint types.
- Updated the hint model to accept all hint types.

### CODE REVIEW

File: src/server/routes/api/document/hint/pubtator.js [hunk 1]
```diff
@@ -12,18 +12,32 @@ function map(bioCDocument) {
...
   });
   const entityTypes = new Map([
+    [PUBTATOR_ANNOTATION_TYPE.GENE, HINT_TYPE.GGP],
     [PUBTATOR_ANNOTATION_TYPE.SPECIES, HINT_TYPE.ORGANISM],
+    [PUBTATOR_ANNOTATION_TYPE.CHEMICAL, HINT_TYPE.CHEMICAL],
+    [PUBTATOR_ANNOTATION_TYPE.DISEASE, HINT_TYPE.DISEASE],
+    [PUBTATOR_ANNOTATION_TYPE.CELL_LINE, HINT_TYPE.CELL_LINE],
   ]);
   const database2Xref = new Map([
+    [PUBTATOR_ANNOTATION_TYPE.GENE, COLLECTIONS.NCBI_GENE],
```
Comment (jvwong) [inline src/server/routes/api/document/hint/pubtator.js:37]: I think this should be `PUBTATOR_DATABASE.ncbi_gene` rathan than `.GENE`

Comment (fileoy) [reply src/server/routes/api/document/hint/pubtator.js:37]: Right, my bad

File: src/model/hint.js
Comment (jvwong) [inline src/model/hint.js:1]: [Last round](https://github.com/PathwayCommons/factoid/pull/1293/commits/d733b4b4ad739f648a3c6ea0016af55c042e2392) you modified the model so the constructor avoids the use of getters and setters (validation). Also in the tests (see below) you avoid their usage, which sorta defeats the purpose of the model.

File: test/pubtator/pubtator.js [hunk 2]
```diff
@@ -77,39 +78,100 @@ describe('pubtator', function () {
...
         });
 
-        it(`Should be all an ORGANISM hint`, function () {
-          const isOrganism = (h) => h.type === HINT_TYPE.ORGANISM;
-          expect(hints.every(isOrganism)).to.be.true;
-        });
-
         it('Should aggregate organism hints correctly', function () {
           const aggregateCounts = hints.reduce((acc, hint) => {
             const key = `${hint._xref.id}_${hint.section}`;
```
Comment (jvwong) [inline test/pubtator/pubtator.js:83]: This avoids the use of the model, which I believe defeats the purposes of the model.

**Review by jvwong — CHANGES_REQUESTED** [review]

**Review by jvwong — APPROVED** [review]
Great merge it!

**Pull request merged**